packages/components/transfer/src/composables/use-check.ts

```diff
@@ -1,53 +1,40 @@
-// @ts-nocheck
-import { computed, getCurrentInstance, watch } from 'vue'
-
-import type { ExtractPropTypes } from 'vue'
-import type { Key, TransferPanelState } from './transfer'
-
-export const CHECKED_CHANGE_EVENT = 'checked-change'
-
-export const useCheckProps = {
-  data: {
-    type: Array,
-    default() {
-      return []
-    },
-  },
-  optionRender: Function,
-  placeholder: String,
-  title: String,
-  filterable: Boolean,
-  format: Object,
-  filterMethod: Function,
-  defaultChecked: Array,
-  props: Object,
-}
+import { computed, watch } from 'vue'
+import { isFunction } from '@element-plus/utils'
+import { CHECKED_CHANGE_EVENT } from '../transfer-panel'
+
+import type { SetupContext } from 'vue'
+import type { CheckboxValueType } from '@element-plus/components/checkbox'
+import type { TransferKey } from '../transfer'
+import type {
+  TransferPanelEmits,
+  TransferPanelProps,
+  TransferPanelState,
+} from '../transfer-panel'
 
 export const useCheck = (
-  props: ExtractPropTypes<typeof useCheckProps>,
-  panelState: TransferPanelState
+  props: TransferPanelProps,
+  panelState: TransferPanelState,
+  emit: SetupContext<TransferPanelEmits>['emit']
 ) => {
-  const { emit } = getCurrentInstance()
-
-  const labelProp = computed(() => props.props.label || 'label')
-
-  const keyProp = computed(() => props.props.key || 'key')
-
-  const disabledProp = computed(() => props.props.disabled || 'disabled')
+  const {
+    label: propsLabel,
+    key: propsKey,
+    disabled: propsDisabled,
+  } = props.props
```

Comment:
There seems to be loss of reactive here.

```diff
- const {
-    label: propsLabel,
-    key: propsKey,
-    disabled: propsDisabled,
-  } = props.props
+ const propsAlias = usePropsAlias(props)
```

The usePropsAlias functions are meant to be use in multiple places.

Reply:
It's my bad, I wrongly thought to call it as few as possible. 😅